pyproject.toml

```diff
@@ -0,0 +1,41 @@
+[tool.ruff]
+select = [
+  "C90",    # McCabe cyclomatic complexity
+  "E",      # pycodestyle
+  "F",      # Pyflakes
+  "ICN",    # flake8-import-conventions
+  "INT",    # flake8-gettext
+  "PLC",    # Pylint conventions
+  "PLE",    # Pylint errors
+  "PLR09",  # Pylint refactoring: max-args, max-branches, max returns, max-statements
+  "PYI",    # flake8-pyi
+  "RSE",    # flake8-raise
+  "RUF",    # Ruff-specific rules
+  "T10",    # flake8-debugger
+  "TCH",    # flake8-type-checking
+  "TID",    # flake8-tidy-imports
+  "W",      # pycodestyle
+  "YTT",    # flake8-2020
+]
+exclude = [
+  "deps",
+  "tools/inspector_protocol",
+]
```

Comment:
This is a smaller list of exclusions than were in the flake8 settings:
```
exclude=.git,deps,lib,src,tools/gyp,tools/inspector_protocol,tools/pip,tools/v8_gypfiles/broken
```

with `tools/gyp` being the obvious difference based on all of the changes to `tools/gyp` in this PR.

Reply:
Why exclude directories that are passing the test?  We have been improving the gyp codebase.
* https://github.com/nodejs/gyp-next/pull/193
* https://github.com/nodejs/node-gyp/pull/2815

Reply:
Because they're not passing the test without the changes in this PR?

If we need to update `tools/gyp` with the latest gyp-next we should do that separately.